From: Werner Koch <wk@gnupg.org>
Date: Thu, 19 Oct 2017 17:12:36 +0200
Subject: gpg: Print sec/sbb with --import-option import-show or show-only.

* g10/import.c (import_one): Pass FROM_SK to list_keyblock_direct.
--

Note that this will likely add the suffix '#' top "sec" because the
secret key has not yet (or will not be) imported.  If the secret key
already exists locally another suffix might be printed.  The upshot is
that the suffix has no usefulness.

GnuPG-bug-id: 3431
Signed-off-by: Werner Koch <wk@gnupg.org>
(cherry picked from commit 2c7dccca9b617780a3ea760adf460bb3b77f90f3)
---
 doc/gpg.texi | 3 ++-
 g10/import.c | 3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/doc/gpg.texi b/doc/gpg.texi
index ca6f53698..9f95c9a6f 100644
--- a/doc/gpg.texi
+++ b/doc/gpg.texi
@@ -2269,7 +2269,8 @@ opposite meaning. The options are:
   @item import-show
   Show a listing of the key as imported right before it is stored.
   This can be combined with the option @option{--dry-run} to only look
-  at keys.
+  at keys. Note that suffixes like '#' for "sec" and "sbb" lines
+  may or may not be printed.
 
   @item import-export
   Run the entire import code but instead of storing the key to the
diff --git a/g10/import.c b/g10/import.c
index bddb2cfc7..a83551c41 100644
--- a/g10/import.c
+++ b/g10/import.c
@@ -1542,7 +1542,8 @@ import_one (ctrl_t ctrl,
       merge_keys_done = 1;
       /* Note that we do not want to show the validity because the key
        * has not yet imported.  */
-      list_keyblock_direct (ctrl, keyblock, 0, 0, 1, 1);
+      list_keyblock_direct (ctrl, keyblock, from_sk, 0,
+                            opt.fingerprint || opt.with_fingerprint, 1);
       es_fflush (es_stdout);
     }
 
